### DISCUSSION TITLE
Yield computation when battery is charged from grid

### DISCUSSION THREAD

**Discussion by ChrisBCH** [wlcrs/huawei_solar#458] [General]
The yield computation was discussed in several threads.
However, I think one point was not taken into account yet: How must the yield be computed when the battery is charged from the grid?
Now as days get darker I had the situation where the battery was loaded from the minimum SOC (15%) to 25% from the grid automatically. This charge is incorrectly added to the yield.
Any ideas how to correct for this?

**Comment by Joebinator** [comment]
Hi!

have you found a solution? I am planing to charge my battery from grid if prices are very low.

**Comment by Roving-Ronin** [comment]
@ChrisBCH which Yield calculation are you using / referring too ? To get a figure that shows the amount of energy put into the LUNA battery and break it down into what came from Solar PV vs Grid Power (vs just the 'daily yield' of the battery from all sources) requires the use of custom/derived sensors to do this.

**Comment by Joebinator** [comment]
Hi!

The sensor.inverter_daily_yield as also the sensor.inverter_total_yield were increasing.

II have configured my sensors according to the wiki (https://github.com/wlcrs/huawei_solar/wiki/Daily-Solar-Yield)

But these also reffer to the yield sensor from the integration, and with them increasing i get those values.

Any ideas or examles (maybe anyone has done this before) how to derive battery charged from solar vs battery charged from grid?


<img width="1303" alt="image" src="https://github.com/user-attachments/assets/1cc96ca0-dabf-4337-92a5-3321442c7fc3">



<img width="1016" alt="image" src="https://github.com/user-attachments/assets/1fc85459-bcd4-495f-a91d-2b2ce7053489">

**Comment by Joebinator** [comment]
i also does increase when there is power from the panels

i have charged once a again from grid due to low energy prices and no sun today

<img width="1325" alt="image" src="https://github.com/user-attachments/assets/78fba3d2-718d-44e1-a493-8741bd60107d">

**Comment by StephanRoemer** [comment]
Same issue here. I was using the formula from the wiki (the last one on the page), but something is not working right here.
